.ci/robot_framework/libs/TestUtils.py

```diff
@@ -15,6 +15,31 @@ class TestUtils:
     def __init__(self):
         self.driver = None
 
+    def get_machine_expectation(self, id, machine, wpeversion, type="text"):
+        # Define file paths to search in order
+        files = [
+                os.path.join("..", "..", "robot_framework", "expectations", wpeversion, f"{machine}.expectations"),
+                os.path.join("..", "..", "robot_framework", "expectations", f"{machine}.expectations"),
+                os.path.join("..", "..", "robot_framework", "expectations", wpeversion, "generic.expectations"),
+                os.path.join("..", "..", "robot_framework", "expectations", "generic.expectations")
+                ]
+
+        for f in files:
+            if os.path.exists(f):
+                with open(f, "r", encoding="utf-8") as fp:
+                    for line in fp:
+                        # Remove comments and whitespace then process line
+                        line = line.split("#", 1)[0].strip()
+                        if not line:
+                            continue
+                        # Each line should be in the format key:value
+                        if ":" in line:
+                            key, value = [x.strip() for x in line.split(":", 1)]
+                            if key == id:
+                                return value
```

Comment:
Alternatively, you can use a regex to parse the line. I think something like this would work:

```python
m = re.match("^(.+?):(.+?)$", line)
if m:
    key, value = m.group(1), m.group(2)
```

It would require to `import re`.